akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/scaladsl/ClusterShardingSpec.scala

```diff
@@ -265,6 +264,37 @@ class ClusterShardingSpec
       charlieRef ! StopPlz()
     }
 
+    "EntityRef - tell (using system.entityRefFor) " in {
+      val typeKeyWithEnvelopes = akka.actor.typed.EntityTypeKey[TestProtocol]("envelope-shard")
+      val charlieRef = system.entityRefFor(typeKeyWithEnvelopes, "charlie")
+
+      val p = TestProbe[String]()
+
+      charlieRef ! WhoAreYou(p.ref)
+      p.receiveMessage() should startWith("I'm charlie")
+
+      charlieRef.tell(WhoAreYou(p.ref))
+      p.receiveMessage() should startWith("I'm charlie")
+
+      charlieRef ! StopPlz()
+    }
+
+    "EntityRef - tell (using system.initEntity system.entityRefFor) " in {
+      val typeKeyWithEnvelopes = akka.actor.typed.EntityTypeKey[TestProtocol]("envelope-shard-2")
+      system.initEntity(akka.actor.typed.Entity(typeKeyWithEnvelopes)(ctx => behavior(ctx.shard)))
+      val charlieRef = system.entityRefFor(typeKeyWithEnvelopes, "charlie")
+
+      val p = TestProbe[String]()
+
+      charlieRef ! WhoAreYou(p.ref)
+      p.receiveMessage() should startWith("I'm charlie")
+
+      charlieRef.tell(WhoAreYou(p.ref))
+      p.receiveMessage() should startWith("I'm charlie")
+
+      charlieRef ! StopPlz()
+    }
+
```

Comment:
I just added this here to move fast. If we go forward with this idea, it must have it's own tests. Probably repeating everything that's done in this test, but using `system.initEntity` and `system.entityRefFor`